mlos_bench/mlos_bench/services/remote/azure/azure_services.py

```diff
@@ -201,6 +206,20 @@ def __init__(self,
             with open(self._custom_data_file, 'r', encoding='utf-8') as custom_data_fh:
                 self._deploy_params["customData"] = custom_data_fh.read()
 
+    def _get_session(self, params: dict) -> requests.Session:
+        total = params.get("requestTotalRetries", self._total_retries)
+        backoff_factor = params.get("requestBackoffFactor", self._backoff_factor)
+
+        session = requests.Session()
+        retry = Retry(total=total, backoff_factor=backoff_factor)
+        adapter = HTTPAdapter(max_retries=retry)
+
+        session.mount("https://", adapter)
+
+        session.headers.update(self._get_headers())
+
+        return session
```

Comment:
I'd like more compact
```suggestion
        total_retries = params.get("requestTotalRetries", self._total_retries)
        backoff_factor = params.get("requestBackoffFactor", self._backoff_factor)
        session = requests.Session()
        session.mount("https://", HTTPAdapter(
            max_retries=Retry(total=total_retries, backoff_factor=backoff_factor)))
        session.headers.update(self._get_headers())
        return session
```
another nitpick, though